### ISSUE TITLE
What does the "centroids" mean?

### ISSUE THREAD

**Issue by 9p15p** [microsoft/transductive-vos.pytorch#12]
Hi, Thank you for your excellent work!

I want to know what the 'annotation_centroids.npy' and the centroids means.
To be specific，I can't find something in paper related to `torch.argmin(torch.sqrt(torch.sum((img.unsqueeze(1) - centroids) ** 2, 2)), 1)` in function 'rgb2class'.

Best wishes.

**Here is my question in Chinese，which is consistent with the above:**

'centroids'这个变量以及在dataset中的 'annotation_centroids.npy' 发挥了什么作用呢，我在论文中没有看到对应的信息。
 关于'rgb2class'函数中的 `torch.argmin(torch.sqrt(torch.sum((img.unsqueeze(1) - centroids) ** 2, 2)), 1)` 具体含义麻烦您能解释一下吗

祝好。

**Comment by wolfLeo** [comment]
annotation is represented by rgb， in order to convert to label， we need a mapping tuple (r, g, b) -> interger (instance id), due to resize， annotation rgb value not exactly in the mapping's domain of definition, so take the nearset, this is why called centroids

**Comment by jili-li** [comment]
> Hi, Thank you for your excellent work!
> 
> I want to know what the 'annotation_centroids.npy' and the centroids means.
> To be specific，I can't find something in paper related to `torch.argmin(torch.sqrt(torch.sum((img.unsqueeze(1) - centroids) ** 2, 2)), 1)` in function 'rgb2class'.
> 
> Best wishes.
> 
> **Here is my question in Chinese，which is consistent with the above:**
> 
> 'centroids'这个变量以及在dataset中的 'annotation_centroids.npy' 发挥了什么作用呢，我在论文中没有看到对应的信息。
> 关于'rgb2class'函数中的 `torch.argmin(torch.sqrt(torch.sum((img.unsqueeze(1) - centroids) ** 2, 2)), 1)` 具体含义麻烦您能解释一下吗
> 
> 祝好。

annotation is represented by rgb， in order to convert to label， we need a mapping tuple (r, g, b) -> interger (instance id), due to resize， annotation rgb value not exactly in the mapping's domain of definition, so take the nearest, this is why called centroids

**Comment by wolfLeo** [comment]
and i have another question, why do you use 191 instead of 192? Thank you?

**Comment by jili-li** [comment]
and i have another question, why do you use 191 instead of 192? Thank you!

**Comment by 9p15p** [comment]
> and i have another question, why do you use 191 instead of 192? Thank you!

Thank you for your reply.^_^
and I don't know what your question about '191' or '192' means.  Do I use it somewhere?

**Comment by jili-li** [comment]
> > and i have another question, why do you use 191 instead of 192? Thank you!
> 
> Thank you for your reply.^_^
> and I don't know what your question about '191' or '192' means. Do I use it somewhere?

sorry， I mean in annotation_centroids.npy they use the value 191， but in palette it's 192

**Comment by 9p15p** [comment]
> > > and i have another question, why do you use 191 instead of 192? Thank you!
> > 
> > 
> > Thank you for your reply.^_^
> > and I don't know what your question about '191' or '192' means. Do I use it somewhere?
> 
> sorry， I mean in annotation_centroids.npy they use the value 191， but in palette it's 192

Oh, I find this problem, too. Thank you.
